tools/toolchains/arm.py

```diff
@@ -353,13 +353,17 @@ def __init__(self, target, notify=None, macros=None,
                  silent=False, extra_verbose=False, build_profile=None,
                  build_dir=None):
         target.default_toolchain = "uARM"
-        ARM.__init__(self, target, notify, macros, build_dir=build_dir,
-                     build_profile=build_profile)
-        if not set(("ARM", "uARM")).intersection(set(target.supported_toolchains)):
+
+        #At this point we already know that we want to use ARMC5+Microlib, so check for if they are supported
+        #For, AC6+Microlib we still use ARMC6 class
+        if not set(("uARM","ARMC5")).issubset(set(target.supported_toolchains)):
```

Comment:
This undermines much of your work above to be backwards compatible with prior version of Mbed OS. This will raise an exception in the case that we don't have `ARMC5` in the supported toolchain list, which is always true for all prior versions of Mbed OS.

Reply:
@theotherjimmy - I have the fixes now, please review.